core/common/type_traits.hpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
```

Comment:
Just two way:
- pragma-once as faster, but non-standard
- include-guard works everywhere, but slower

Reply:
I think the last time we discussed it we came to the conclusion that since we already have only include-guards in the majority of our project and adding or replacing with pragmas does not result in any significant benefit, it's better to leave just common include guards. I'd rather stick to that, although your approach indisputably makes sense.

Reply:
As you wish. I was removed in acconding convention